### PR TITLE
Safari fix: re-use existing AudioContext

### PR DESCRIPTION
**Pull request by alexthewilde**
Safari puts a limit on the amount of AudioContexts that can be created. An application that instantiates the recorder multiple times must therefore be able to re-use an existing AudioContext instance.

### CODE REVIEW

File: src/recorder.ts [hunk 1]
```diff
@@ -36,11 +36,10 @@ export const getMp3MediaRecorder = (config: RecorderConfig): Promise<typeof Medi
 
         static isTypeSupported = (mimeType: string) => mimeType === MP3_MIME_TYPE;
 
-        constructor(stream: MediaStream) {
+        constructor(stream: MediaStream, audioContext?: AudioContext) {
```
Comment (eliasmeire) [inline src/recorder.ts]: MediaRecorder constructor already has a second `options` argument. To keep this ponyfill fully compatible (and for extendability) I would like to keep the second argument as an object. Could you make the second argument an object with a property audioContext?

**Comment by eliasmeire** [comment]
@alexthewilde I can also do the changes if you want, there should be a "Allow edits from maintainers" checkbox somewhere in the right sidebar.

**Comment by alexthewilde** [comment]
@eliasmeire yes please! You should be able to make changes (that box has been checked already).

**Pull request merged**